src/components/Tab/Tab.tsx

```diff
@@ -52,7 +61,27 @@ export const Tab = ({
       role="tabpanel"
       {...other}
     >
-      {children}
+      {withoutTypes(children, TabButton)}
     </div>
   );
 };
+
+const TabButton = ({ children }: TabButtonProps) => {
+  const context = useContext(TabsContext);
+  return (
+    <div>{typeof children === 'function' ? children(context) : children}</div>
```

Comment:
Could `<Tab>` take the render prop instead? Or do we need the additional component/div layer?

Reply:
I created `Tab.Button` to mimic the structure of some of our other components with render props, like `Accordion` and a few of the other HeadlessUI offshoots. But in testing this I found something PRETTY peculiar. I am gonna submit a simplified update to this PR to demonstrate what i mean.

Besides that, my plan is to eventually add something like a `Tab.Panel` to contain the non-button children to make the rest of the API match our other components:

```jsx
<Tab title="Some Title">
    <Tab.Button>...</Tab.Button>
    <Tab.Panel>...</Tab.Panel>
</Tab>
```

Reply:
I think, long-term, the final step to make all these internally consistent would be to add the render prop to `Tab` as you're saying, and it would contain `Tab.Button`s and `Tab.Panel`s. Based on how `Tabs` is implemented, it might be a breaking change. I'm for it, tho... the component is a little weird right now.